### PR TITLE
Move all module settings to Configuration > Settings

### LINKED ISSUE #11773: Move Security Settings into Configuration Section
I just watched the latest OC standup, @sebastienros pointed why there's a new top level menu for Security Settings? I already though about it two days ago, so indeed we need to move it to configuration like other modules

### LINKED ISSUE #11942: Move all settings menu items under Configuration > Settings menu
### Is your feature request related to a problem? Please describe.
Currently, site settings provided by modules are scattered on the menu in many places. Form user experience, it would be nice to place all settings under one dedicated location on the menu for easy of access and quickly know where are all the settings located. 

Also, the text style on the menu items isn’t unified.

### Describe the solution you'd like
Move any site setting or configurations under the “Configuration > Settings” menu

Additionally, we may want to unify the naming by using a common style. Either a title case or sentence case for every menu item. I think using sentence style will look good on the menu item as per the [recommendation here](https://www.patternfly.org/v4/ux-writing/capitalization/#capitalization-across-patternfly)

Here is how the new menu structure may look like

Configuration > Settings > Search
Configuration > Settings > Security > User login
Configuration > Settings > Security > User change email
Configuration > Settings > Security > User registration
Configuration > Settings > Security > User reset password
Configuration > Settings > Security > Security header
Configuration > Settings > Security > HTTPS
Configuration > Settings > Security > CORS
Configuration > Settings > Security > reCaptcha
Configuration > Settings > Security > OpenID Connect > Authentication client
Configuration > Settings > Security > OpenID Connect > Authorization server
Configuration > Settings > Security > OpenID Connect > Token validation
Configuration > Settings > Security > OpenID Connect > Management
Configuration > Settings > Security > Authentication > Facebook
Configuration > Settings > Security > Authentication > Twitter
Configuration > Settings > Security > Authentication > GitHub
Configuration > Settings > Security > Authentication > Google
Configuration > Settings > Security > Authentication > Microsoft
Configuration > Settings > Security > Authentication > Azure Active Directory

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #11942 
Fix #11773
Closes #11972

Move all settings menu items into to one central place (i.e., `Configuration > Settings`)

Also, make the spacing/formatting consistent in all of the AdminManu classes for consistency and a bit more better readability.

Here is how the menu looks when all the feature the provide settings are enabled

image

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             }
 
             // Don't add the menu item on non-default tenants
-            if (_shellSettings.Name != ShellHelper.DefaultShellName)
+            if (!_shellSettings.IsDefaultShell())
```
Comment (CrestApps) [inline src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs:28]: Why do we need this line in this code? If this class is only registered when the tenants module is enabled and the tenant module is only available to the default tenant. Is this something left due to old logic or should it stay as part assurance measure?

File: src/OrchardCore.Modules/OrchardCore.Twitter/AdminMenu.cs [hunk 2]
```diff
@@ -18,17 +18,28 @@ public AdminMenuSignin(IStringLocalizer<AdminMenuSignin> localizer)
```
Comment (CrestApps) [inline src/OrchardCore.Modules/OrchardCore.Twitter/AdminMenu.cs:18]: Why this class and few other are decorated with `[Feature]` attribute (i.e., `[Feature(TwitterConstants.Features.Signin)]`?) Each of these classes get registered only when the corresponding feature is enabled (in this case "OrchardCore.Twitter.Signin"). I think adding the `[Feature]` attribute here is redundant. `[Feature]` attribute is helpful in some scenarios like a controller, but not sure how it is helpful here?

**Review by CrestApps — COMMENTED** [review]
I have some question. I would appreciate any feedback to see if more cleaning is needed or learn something new.

**Comment by hishamco** [comment]
Please don't request the review from many unless there's a reason for it, regarding the PR I forgot to push my PR regarding the issue I created, but it's already on my local. For the new settings changes that you made let us see if it's fine to move them into configuration menu

Last thing please don't change anything else unrelated to PR, I appreciate the effort you did, bu it's good to focus on fixing the issue itself

Thanks

**Comment by Skrypt** [comment]
@hishamco The request for review to multiple people was automatic as Github added them as "code owners".

**Comment by hishamco** [comment]
I see, but I think you can't remove them before push the PR

File: src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs [hunk 3]
```diff
@@ -23,11 +23,12 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             }
 
             builder
-                .Add(S["Audit Trail"], NavigationConstants.AdminMenuAuditTrailPosition, configuration => configuration
+                .Add(S["Audit trail"], NavigationConstants.AdminMenuAuditTrailPosition, audit => audit
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs:26]: Out of scope, and these changes are actually breaking the translations

Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs:26]: @sebastienros what is out of scope changing `S["Audit Trail"]` to `S["Audit trail"]` or standardizing how the menu items appear on the menu?

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs:26]: Changing the menu names break translations. So, he suggests making a second PR for that matter since it's a different scope than the actual PR.

Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs]: I see. I has been resolved. I think we should create another issue for creating consistent formatting including defining the standards.

**Comment by hishamco** [comment]
@CrestApps I will push a PR for #11773 and merge it then, I will keep this open because what I saw there's a breaking changes in the translation, so please fix the conflict one I merge my PR

Thanks

File: src/OrchardCore.Modules/OrchardCore.AdminMenu/Services/AdminMenuNavigationProvidersCoordinator.cs [hunk 4]
```diff
@@ -44,7 +44,7 @@ public async Task BuildNavigationAsync(string name, NavigationBuilder builder)
                 return;
             }
 
-            var trees = ((await _adminMenuService.GetAdminMenuListAsync()).AdminMenu)
+            var trees = (await _adminMenuService.GetAdminMenuListAsync()).AdminMenu
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.AdminMenu/Services/AdminMenuNavigationProvidersCoordinator.cs:47]: This is unrelated to the PR

File: src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs [hunk 5]
```diff
@@ -22,9 +22,9 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             }
 
             builder
-                .Add(S["Configuration"], NavigationConstants.AdminMenuConfigurationPosition, design => design
+                .Add(S["Configuration"], NavigationConstants.AdminMenuConfigurationPosition, configuration => configuration
                     .AddClass("menu-configuration").Id("configuration")
-                    .Add(S["GraphiQL"], S["GraphiQL"].PrefixPosition(), deployment => deployment
+                    .Add(S["GraphiQL"], S["GraphiQL"].PrefixPosition(), graphiQl => graphiQl
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: graphQL

File: src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs [hunk 6]
```diff
@@ -23,11 +23,12 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             }
 
             builder
-                .Add(S["Audit Trail"], NavigationConstants.AdminMenuAuditTrailPosition, configuration => configuration
+                .Add(S["Audit Trail"], NavigationConstants.AdminMenuAuditTrailPosition, audit => audit
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.AuditTrail/Navigation/AuditTrailAdminMenu.cs:26]: auditTrail

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/AdminMenu.cs [hunk 7]
```diff
@@ -25,16 +25,16 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
...
                         .Add(S["Localization"], localization => localization
-                            .Add(S["Content Request Culture Provider"], S["Content Request Culture Provider"].PrefixPosition(), registration => registration
+                            .Add(S["Content Request Culture Provider"], S["Content Request Culture Provider"].PrefixPosition(), provider => provider
                                 .AddClass("contentrequestcultureprovider").Id("contentrequestcultureprovider")
                                 .Action("Index", "Admin", new { area = "OrchardCore.Settings", groupId = ContentRequestCultureProviderSettingsDriver.GroupId })
                                 .Permission(Permissions.ManageContentCulturePicker)
                                 .LocalNav()
                             )
-                            .Add(S["Content Culture Picker"], S["Content Culture Picker"].PrefixPosition(), registration => registration
+                            .Add(S["Content Culture Picker"], S["Content Culture Picker"].PrefixPosition(), picker => picker
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/AdminMenu.cs:37]: culturePicker or contentCulturePicker

File: src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs [hunk 8]
```diff
@@ -48,13 +48,12 @@ public class AdminMenu : INavigationProvider
...
         public async Task BuildNavigationAsync(string name, NavigationBuilder builder)
         {
-            var context = _httpContextAccessor.HttpContext;
-
             if (!String.Equals(name, "admin", StringComparison.OrdinalIgnoreCase))
             {
                 return;
             }
 
+            var context = _httpContextAccessor.HttpContext;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs:56]: Un related to the PR

File: src/OrchardCore.Modules/OrchardCore.Demo/AdminMenu.cs [hunk 9]
```diff
@@ -26,26 +26,33 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
...
                     .Add(S["This Menu Item 3"], "0", item => item
                         .Add(S["This is Menu Item 3.1"], subItem => subItem
-                            .Action("Index", "Admin", new { area = "OrchardCore.Demo" }))
+                            .Action("Index", "Admin", new { area = "OrchardCore.Demo" })
+                        )
                         .Add(S["This is Menu Item 3.2"], subItem => subItem
-                            .Action("Index", "Admin", new { area = "OrchardCore.Demo" }))
-
+                            .Action("Index", "Admin", new { area = "OrchardCore.Demo" })
+                        )
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Demo/AdminMenu.cs:55]: Extra whitespace

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/AdminMenu.cs [hunk 10]
```diff
@@ -24,12 +24,12 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             builder
                 .Add(S["Configuration"], configuration => configuration
                     .Add(S["Import/Export"], import => import
-                        .Add(S["Remote Instances"], S["Remote Instances"].PrefixPosition(), remote => remote
+                        .Add(S["Remote Instances"], S["Remote Instances"].PrefixPosition(), instances => instances
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment.Remote/AdminMenu.cs:27]: remoteInstances

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/AdminMenu.cs [hunk 11]
```diff
@@ -24,12 +24,12 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
...
                             .Action("Index", "RemoteInstance", new { area = "OrchardCore.Deployment.Remote" })
                             .Permission(Permissions.ManageRemoteInstances)
                             .LocalNav()
                         )
-                        .Add(S["Remote Clients"], S["Remote Clients"].PrefixPosition(), remote => remote
+                        .Add(S["Remote Clients"], S["Remote Clients"].PrefixPosition(), clients => clients
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment.Remote/AdminMenu.cs:32]: remoteClients

**Review by hishamco — CHANGES_REQUESTED** [review]
I'm on half of the reviewing the changes ;) just few notes, please remove unrelated changes to this PR. Fix formating and remove extra whitespaces

**Comment by CrestApps** [comment]
> I'm on half of the reviewing the changes ;) just few notes, please remove unrelated changes to this PR. Fix formating and remove extra whitespaces

I am going to await the outcome of the outcome of the conversation in the issue #11942. I we'll make the necessary changes then. 
On a side note, some of the "unrelated" changes you suggested is just general clean up just to unify the code in all of the AdminMenu classes.

File: src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs (hunk 5 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: graphiQl is fine here. Though, the changes you are requesting are not an issue.

 if it's just about renaming some variables we are getting again, out of scope. This can be done in a different PR.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: In fact I think those are all `menuItem`.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: If that makes any sense here some example that would still work : 

```C#
            builder
                .Add(S["Configuration"], NavigationConstants.AdminMenuConfigurationPosition, navigationItemBuilder => navigationItemBuilder
                    .AddClass("menu-configuration").Id("configuration")
                    .Add(S["GraphiQL"], S["GraphiQL"].PrefixPosition(), navigationItemBuilder => navigationItemBuilder
                        .Action("Index", "Admin", new { area = "OrchardCore.Apis.GraphQL" })
                        .Permission(Permissions.ExecuteGraphQL)
                        .LocalNav()
                    )
                );
```

So, not sure renaming these variables will have any consequence.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: I think we should have simply have used something like "nib" as a convention just like we often use "x => x.something"

Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: These are all builder. I think explicitly naming thing would help when nesting menu items. I tried to make the name relevant while being short. But I don’t know that full naming is really necessary.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: Only if you need to make mutations on the parent navigationItemBuilder from a child. Which probably never happens in OC.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: Yes, but these are builders so you will never need to use them other than for calling methods on them.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/AdminMenu.cs:27]: Anyways, while it is an interesting discussion, we digress. 😄

File: src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs (hunk 8 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Contents/AdminMenu.cs:56]: no incidence

File: src/OrchardCore.Modules/OrchardCore.AdminMenu/Services/AdminMenuNavigationProvidersCoordinator.cs (hunk 4 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.AdminMenu/Services/AdminMenuNavigationProvidersCoordinator.cs:47]: no incidence

File: src/OrchardCore.Modules/OrchardCore.Localization/AdminMenu.cs [hunk 12]
```diff
@@ -25,23 +25,25 @@ public AdminMenu(IStringLocalizer<AdminMenu> localizer)
         ///<inheritdocs />
         public Task BuildNavigationAsync(string name, NavigationBuilder builder)
         {
-            if (String.Equals(name, "admin", StringComparison.OrdinalIgnoreCase))
+            if (!String.Equals(name, "admin", StringComparison.OrdinalIgnoreCase))
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Localization/AdminMenu.cs:28]: I'm not sure doing this will improve execution time since it needs to evaluate that "if" still. But for code consistency, I guess it is not bad either.

File: src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs (hunk 1 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs:28]: It is fine. Only the first "default" tenant should be able to create sub-tenants, that's why.

File: src/OrchardCore.Modules/OrchardCore.Localization/AdminMenu.cs (hunk 12 above)
Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.Localization/AdminMenu.cs:28]: I don’t think it’ll effect execution time. The statement is still have to be evaluated. It just consistent and reduce indentation

File: src/OrchardCore.Modules/OrchardCore.Twitter/AdminMenu.cs (hunk 2 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Twitter/AdminMenu.cs:18]: Question for @jtkech
I think these are redundant too. Though, maybe there is a reason.

**Review by Skrypt — APPROVED** [review]

File: src/OrchardCore.Modules/OrchardCore.Security/AdminMenu.cs [hunk 13]
```diff
@@ -17,16 +17,23 @@ public AdminMenu(IStringLocalizer<AdminMenu> localizer)
...
-                    .AddClass("security").Id("security")
-                        .Add(S["Settings"], settings => settings
+                return Task.CompletedTask;
+            }
+
+            builder
+                .Add(S["Configuration"], configuration => configuration
+                    .Add(S["Settings"], settings => settings
+                        .Add(S["Security"], security => security.Id("security")
                             .Add(S["Security Headers"], S["Security Headers"].PrefixPosition(), headers => headers
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Security/AdminMenu.cs:29]: Here is the same change from https://github.com/OrchardCMS/OrchardCore/pull/11972/files

File: src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs (hunk 1 above)
Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs:28]: Yes I get it. Be the module is only available to host tenant which you can only have one. So I think this code is redundant. I mean even if we remove it this class will only be called by the host tenant. Not sure what I am missing here but

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs:28]: It is redundant, though, at some point, it was a necessity. Maybe it could be removed but I would not guess it for now. Leave it there.

File: src/OrchardCore.Modules/OrchardCore.Security/AdminMenu.cs (hunk 13 above)
Comment (CrestApps) [reply src/OrchardCore.Modules/OrchardCore.Security/AdminMenu.cs:29]: PR #11972 was closed so the change exists only here.

**Comment by Skrypt** [comment]
image

**Comment by MikeAlhayek** [comment]
@sebastienros maybe this is something we should discuss on Tuesday to either close it or approve the change... or request a vote on.

**Comment by agriffard** [comment]
Current menus with settings:
image
image
image
image

**Comment by urbanit** [comment]
tbh, I expect search settings to be under "Search Module", Security Settings to be under Security, Localization Settings under Localization module etc...
This way custom modules will "keep" their settings inside module area and would be easier for user to locate them
What is proposed, I am afraid it will duplicate the modules structure under "configuration">"settings"

**Comment by sebastienros** [comment]
One down vote, nobody else cares.
